packages/shell-api/src/streams.ts

```diff
@@ -0,0 +1,174 @@
+import type { Document } from 'bson';
+import { CommonErrors, MongoshInvalidInputError } from '@mongosh/errors';
+
+import { MongoshInterruptedError } from './interruptor';
+import {
+  returnsPromise,
+  shellApiClassDefault,
+  ShellApiWithMongoClass,
+} from './decorators';
+import StreamProcessor from './stream-processor';
+import { ADMIN_DB, asPrintable, shellApiType } from './enums';
+import type Database from './database';
+import type Mongo from './mongo';
+
+@shellApiClassDefault
+export class Streams extends ShellApiWithMongoClass {
+  public static newInstance(database: Database) {
+    return new Proxy(new Streams(database), {
+      get(target, prop) {
+        const v = (target as any)[prop];
+        if (v !== undefined) {
+          return v;
+        }
+        if (typeof prop === 'string' && !prop.startsWith('_')) {
+          return target.getProcessor(prop);
+        }
+      },
+    });
+  }
+
+  private _database: Database;
+
+  constructor(database: Database) {
+    super();
+    this._database = database;
+  }
+
+  get _mongo(): Mongo {
+    return this._database._mongo;
+  }
+
+  [asPrintable](): string {
+    return 'Atlas Stream Processing';
+  }
+
+  getProcessor(name: string) {
+    return new StreamProcessor(this, name);
+  }
+
+  @returnsPromise
+  async process(pipeline: Document[], options?: Document) {
+    if (!Array.isArray(pipeline) || !pipeline.length) {
+      throw new MongoshInvalidInputError(
+        'Invalid pipeline',
+        CommonErrors.InvalidArgument,
+        pipeline
+      );
+    }
+    const result = await this._runStreamCommand({
+      processStreamProcessor: pipeline,
+      ...(options ? { options } : {}),
```

Comment:
Has this code been added for future compatibility? As of now, `processStreamProcessor` enforces that no extra fields are specified: https://github.com/10gen/mongohouse/blob/77426dee9b976d20d4a7fc32baffd71bf3acb6e0/internal/mongoserver/command/streams_parser.go#L71C1-L73C3

```
AtlasStreamProcessing> sp.process($pipeline, {limit:5})
MongoServerError: "options" is not a valid argument to processStreamProcessor, correlationID = 177e143da46c62a1befa0aad
AtlasStreamProcessing>
```

Reply:
Yes, for now. I believe process command internally call create and then start it immediately. And `createStreamProcessor` needs to support passing in `options` according to the [syntax document](https://docs.google.com/document/d/1dC9ebvAHLjMyqStlhEyA0O2pxIceS0r1xWAfFAIrppo/edit#heading=h.m3fnu0kzh5i6).

Reply:
I think it’s a good idea to have some forward compatibility here, even if the server validates that extra options will be rejected for now :+1: